step/step_test.go

```diff
@@ -31,6 +34,37 @@ func Test_SSHPrivateKeyRemoved(t *testing.T) {
 	osEnvManager.AssertCalled(t, "Unset", envKey)
 }
 
+func Test_ErrorRaisedIfSSHAddFails(t *testing.T) {
+	osEnvRepository := createOsEnvRepositoryWithSSHKey()
+	osEnvManager := createEnvmanEnvRepository()
+	fileWriter := createFileWriter()
+	logger := createLogger()
+	tempDirProvider := createTempProvider()
+
+	mockCommand := new(MockCommand)
```

Comment:
I think createCommandWithXYZ method would improve readability

Reply:
Also it would make sense to extract the logic here and the createCommand to have a "configurable" builder so code duplication is avoided.

Reply:
Is it possible to do in Go? The problem is that we need to do execute different `MockCommand.On()` calls in each test case on the mock instance. I don't think it's possible to create a function that takes a lambda parameter that would allow this. In Kotlin, for example, I know this is possible thanks to lambda receivers:

```kotlin
fun createMockCommand(actions: MockCommand.() -> Unit) {
  val command = MockCommand()
  command.actions()
  return command
}

val mockCommand = createMockCommand {
  on("run", mock.Anything).return(null)
  on("setStdOut", mock.Anything).return(null)
}

```

Reply:
https://stackoverflow.com/a/61159213

You can use this to mock multiple values in a sequence.

Reply:
But I'm generally not against repetition **in test code**. It makes easier to understand what's going on in each test case if I don't have to jump from function to function.